src/common/types.ts

```diff
@@ -2,6 +2,7 @@ export interface CommonConfig {
     prefixText?: string | boolean,
```

Comment:
I think it would be useful to define an interface for the logger. In that way, users will know that they should provide a logger with a `.info` function.
What do you think?

Reply:
We could say that the logger has to conform to this interface:

```
/**
 * A Logger conforming at least to the most frequently used methods specified by RFC5424
 */
export interface Logger {
    error(msg: string): void;
    warning(msg: string): void;
    info(msg: string): void;
    debug(msg: string): void;
}
```
([as Winston and many others do](https://github.com/winstonjs/winston#logging-levels)) but then we got the problem of making `console.log` fit in that scheme, people using custom loggers, etc...

All in all, this is JS, I don't think we should be too restrictive...we could certainly either leave it as-is or add this `Logger` interface but with only the method we're using here (i.e. `info()`). I'll wait to hear @hg-pyun 's opinion

Reply:
The console can fit into that interface. We just need to change the `warning` to `warn` (same as Winston). 
Console has the other methods as well (info, log, warn, debug, error...)

I agree that, since this is JS, it doesn't matter too much, but since part of this is written in TS, maybe we can enforce that for the ones who use this in typescript. Not sure tho.

Reply:
Thank you for ideas! As for my idea, I wrote it down in the requested changes below. I think it's better to let the logger function itself run.